### PR TITLE
V3 audio refactoring

### PR DESCRIPTION
**Pull request by sheenobu**

### CODE REVIEW

File: audio/play.go [hunk 1]
```diff
@@ -22,94 +21,162 @@ var PlaybackStartTimeout = 1 * time.Second
...
+	pb, err := PlayAsync(ctx, bus, p, mediaURI)
+	if err != nil {
+		return err
+	}
+	defer pb.Cancel()
+
+	select {
+	case <-pb.StopCh():
+	case <-ctx.Done():
+		if pb.Err() == nil {
```
Comment (Ulexus) [inline audio/play.go]: ctx.Error() here, yes?

Comment (sheenobu) [inline audio/play.go]: I guess that would be more straightforward.

File: audio/play.go [hunk 2]
```diff
@@ -22,94 +21,159 @@ var PlaybackStartTimeout = 1 * time.Second
...
+
+type timeoutErr struct {
+	msg string
+}
+
+func (err timeoutErr) Error() string {
+	return err.msg
+}
+
+func (err timeoutErr) IsTimeout() bool {
```
Comment (Ulexus) [inline audio/play.go]: This kind of test is better implemented as `IsTimeoutError(err error) bool`.  If you can execute the method, you already know it is a timeoutErr.

Comment (sheenobu) [inline audio/play.go]: @Ulexus I was following http://dave.cheney.net/2014/12/24/inspecting-errors . The places where Timeout is checked use their own type TimeoutError interface { IsTimeout() bool }, specifically in tests right now. I'll rename to 'Timeout' though to match all the other APIs.

Comment (Ulexus) [inline audio/play.go]: That'll work; thanks

**Pull request merged**